### PR TITLE
Add the plugin for Apache CXF 3.x

### PR DESCRIPTION
**Pull request by zifeihan**
### Add an agent plugin to support Apache CXF 3.x
- [x] Add a test case for the new plugin, refer to [the doc](https://github.com/apache/skywalking/blob/master/docs/en/guides/Plugin-test.md)
- [x] Add a component id in [the component-libraries.yml](https://github.com/apache/skywalking/blob/master/oap-server/server-bootstrap/src/main/resources/component-libraries.yml)
- [x] Add a logo in [the UI repo](https://github.com/apache/skywalking-rocketbot-ui/tree/master/src/views/components/topology/assets)

Fix https://github.com/apache/skywalking/issues/2562, add an agent plugin to support Apache CXF 3.x, like
image

### CODE REVIEW

File: test/plugin/run.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ force_build="off"
 cleanup="off"
 debug_mode=
 
-mvnw=${home}/../../mvnw
+mvnw=mvn
```
Comment (dmsolr) [inline test/plugin/run.sh:27]: wrong hand?

File: apm-sniffer/apm-sdk-plugin/cxf-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/cxf/v3/client/MessageSenderInterceptorInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        Tags.HTTP.METHOD.set(span, httpRequestMethod);
+        Tags.URL.set(span, generateRequestURL(messageInfo, address));
+        SpanLayer.asRPCFramework(span);
+
+        Map protocolHeaders = (Map) message.get(Message.PROTOCOL_HEADERS);
+
+        CarrierItem next = contextCarrier.items();
+        while (next.hasNext()) {
+            next = next.next();
+            List<String> headerList = new ArrayList<>();
```
Comment (dmsolr) [inline apm-sniffer/apm-sdk-plugin/cxf-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/cxf/v3/client/MessageSenderInterceptorInterceptor.java:70]: suggest: new ArrayList<>(1)

File: test/plugin/run.sh (hunk 1 above)
Comment (zifeihan) [reply test/plugin/run.sh]: > wrong hand?

sorry, fixed.

File: apm-sniffer/apm-sdk-plugin/cxf-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/cxf/v3/client/MessageSenderInterceptorInterceptor.java (hunk 2 above)
Comment (zifeihan) [reply apm-sniffer/apm-sdk-plugin/cxf-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/cxf/v3/client/MessageSenderInterceptorInterceptor.java]: done.

**Comment by wu-sheng** [comment]
Hi, I have a question. By reviewing your screenshot, it seems there are 2 entry spans in your case? 
image

Could you confirm what is the real case?

File: test/plugin/scenarios/cxf-scenario/config/expectedData.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+segmentItems:
+  - serviceName: cxf-scenario
+    segmentSize: ge 3
+    segments:
+      - segmentId: not null
+        spans:
+          - operationName: /CXF/AsyncInvoke
```
Comment (wu-sheng) [inline test/plugin/scenarios/cxf-scenario/config/expectedData.yaml:22]: This async invoke could be in the same thread? I found out you have `capture/continued`, but in the test, this is in the same thread

**Review by wu-sheng — COMMENTED** [review]
Please add necessary comments for your codes.

File: test/plugin/scenarios/cxf-scenario/config/expectedData.yaml (hunk 3 above)
Comment (zifeihan) [reply test/plugin/scenarios/cxf-scenario/config/expectedData.yaml]: fixed.

**Comment by zifeihan** [comment]
> Please add necessary comments for your codes.

done.

**Review by dmsolr — APPROVED** [review]
LGTM

File: test/plugin/scenarios/cxf-scenario/config/expectedData.yaml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      - segmentId: not null
+        spans:
+          - operationName: /cxf-scenario/soap/user
+            operationId: 0
+            parentSpanId: -1
+            spanId: 0
+            spanLayer: Http
+            startTime: nq 0
+            endTime: nq 0
+            componentId: not null
```
Comment (wu-sheng) [inline test/plugin/scenarios/cxf-scenario/config/expectedData.yaml:29]: What is the value of this ID?

File: .github/workflows/plugins-test.0.yaml [hunk 5]
```diff
@@ -70,6 +70,7 @@ jobs:
           - gson-scenario
           - elasticjob-3.x-scenario
           - springmvc-reactive-scenario
+          - cxf-scenario
```
Comment (kezhenxu94) [inline .github/workflows/plugins-test.0.yaml:73]: Better move this to `pugins-test.1.yaml` or `plugins-test.2.yaml`.


```shell
$ python3 tools/select-group.py
/Volumes/data/developer/skywalking/.github/workflows/plugins-test.1.yaml 81
/Volumes/data/developer/skywalking/.github/workflows/plugins-test.2.yaml 81
/Volumes/data/developer/skywalking/.github/workflows/plugins-test.3.yaml 95
/Volumes/data/developer/skywalking/.github/workflows/plugins-test.0.yaml 118
```

File: test/plugin/scenarios/cxf-scenario/config/expectedData.yaml (hunk 4 above)
Comment (zifeihan) [reply test/plugin/scenarios/cxf-scenario/config/expectedData.yaml]: fixed.

File: .github/workflows/plugins-test.0.yaml (hunk 5 above)
Comment (zifeihan) [reply .github/workflows/plugins-test.0.yaml]: fixed.

**Review by wu-sheng — APPROVED** [review]
LGTM.

**Review by kezhenxu94 — APPROVED** [review]

**Pull request merged**